piptools/repositories/pypi.py

```diff
@@ -132,123 +124,51 @@ def find_best_match(self, ireq, prereleases=None):
         if not matching_candidates:
             raise NoCandidateFound(ireq, all_candidates, self.finder)
 
-        if PIP_VERSION < (19, 1):
-            best_candidate = max(
-                matching_candidates, key=self.finder._candidate_sort_key
-            )
-        elif PIP_VERSION < (19, 2):
-            evaluator = self.finder.candidate_evaluator
-            best_candidate = evaluator.get_best_candidate(matching_candidates)
-        elif PIP_VERSION < (19, 3):
-            evaluator = self.finder.make_candidate_evaluator(ireq.name)
-            best_candidate = evaluator.get_best_candidate(matching_candidates)
-        else:
-            evaluator = self.finder.make_candidate_evaluator(ireq.name)
-            best_candidate_result = evaluator.compute_best_candidate(
-                matching_candidates
-            )
-            best_candidate = best_candidate_result.best_candidate
-
-        if PIP_VERSION[:2] <= (19, 3):
-            best_candidate_name = best_candidate.project
-        else:
-            best_candidate_name = best_candidate.name
+        evaluator = self.finder.make_candidate_evaluator(ireq.name)
+        best_candidate_result = evaluator.compute_best_candidate(matching_candidates)
+        best_candidate = best_candidate_result.best_candidate
 
         # Turn the candidate into a pinned InstallRequirement
         return make_install_requirement(
-            best_candidate_name,
+            best_candidate.name,
             best_candidate.version,
             ireq.extras,
             constraint=ireq.constraint,
         )
 
     def resolve_reqs(self, download_dir, ireq, wheel_cache):
-        results = None
-
-        if PIP_VERSION < (10,):
-            reqset = RequirementSet(
-                self.build_dir,
-                self.source_dir,
+        with get_requirement_tracker() as req_tracker, TempDirectory(
+            kind="req-tracker"
```

Comment:
```suggestion
            kind="resolver"
```
Better to give it a proper name

Reply:
Addressed in 2d1cb8c.